### PR TITLE
removeLink: refactor and add tests

### PR DESCRIPTION
**Pull request by ahmetb**
- removeLink is now separated from plugin logic. moved plugin related argument
  prep to call-site.
- removeLink now fails explicitly for non-symlinks. since bin/ is supposed to
  be only symlinks, this sounds appropriate
- added test cases
- handled an ignored removeLink error in `installation.Remove()` method (this is supposed to be ok since we use the same symlink to resolve the plugin installation, so plugin is probably there...)

### CODE REVIEW

File: pkg/installation/install.go [hunk 1]
```diff
@@ -143,12 +147,20 @@ func createOrUpdateLink(binDir string, binary string, plugin string) error {
...
+func removeLink(path string) error {
+	if fi, err := os.Lstat(path); err == nil {
+		if fi.Mode()&os.ModeSymlink == 0 {
+			return fmt.Errorf("file %q is not a symlink (mode=%s)", path, fi.Mode())
+		}
+		if err := os.Remove(path); err != nil {
+			return fmt.Errorf("failed to remove the symlink in %q, err: %v", path, err)
+		}
+		glog.V(3).Infof("Removed symlink from %q", path)
+	} else if !os.IsNotExist(err) {
```
Comment (lbb) [inline pkg/installation/install.go]: Can we please invert the case. !...Not..() is quite difficult to understand

File: pkg/installation/install.go [hunk 2]
```diff
@@ -143,12 +147,20 @@ func createOrUpdateLink(binDir string, binary string, plugin string) error {
...
 }
 
-func removeLink(binDir string, plugin string) error {
-	dst := filepath.Join(binDir, pluginNameToBin(plugin))
-	if err := os.Remove(dst); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("failed to remove the symlink in %q, err: %v", dst, err)
-	} else if err == nil {
-		glog.V(3).Infof("Removed symlink from %q", dst)
+// removeLink removes a symlink reference if exists.
+func removeLink(path string) error {
```
Comment (lbb) [inline pkg/installation/install.go]: I don't think we need a generic remove link function. Can we keep the old function interface? `removeLink(binDir string, plugin string)`, that makes it easier to use.

Comment (ahmetb) [reply pkg/installation/install.go]: @lbb That is really not clean code and vastly reduces testability.

We call `filepath.Join(bin, pluginNameToBin(name))` in so many unrelated places  it's at it is pointing to code smell.

File: pkg/installation/install.go (hunk 1 above)
Comment (ahmetb) [reply pkg/installation/install.go]: Agreed we can unfold and revert this if a bit.

**Pull request merged**